pkg/kv/kvserver/scheduler.go

```diff
@@ -272,6 +279,17 @@ func (s *raftScheduler) worker(ctx context.Context) {
 			s.mu.cond.Wait()
 		}
 
+		// TODO(josh): I have no idea if this is a reasonable way to actually
+		// implement the cordoning!!!
+		// TODO(josh): Let's say a single replica of three is cordoned. Will the
+		// per-replica proposal quota hold back writes to the range? That is not
+		// desirable! Some relevant discussion:
+		// https://github.com/cockroachdb/cockroach/issues/77251
+		if s.cordoned[id] {
```

Comment:
I *think* the destroyStatus is a good way to streamline this, as "destroyed" (for a suitable definition, which will need to be expanded) are never raft handled:

https://github.com/cockroachdb/cockroach/blob/33fe9d15b2660cc40d255a3d448d88bb35c2c39d/pkg/kv/kvserver/replica_raft.go#L1508-L1514

Instead of `Removed()` we'd want a method `Usable() error` or something, which would either return `nil`, `errRemoved`, or `errCordoned`.